python/taichi/lang/ast/ast_transformer_utils.py

```diff
@@ -160,6 +180,9 @@ def is_in_static_for(self):
     def is_in_non_static(self):
         return self.non_static_status.is_in_non_static_scope
 
+    def is_in_static_scope(self):
```

Comment:
Why do we have both `StaticScopeGuard` and `NonStaticScopeGuard`?

Reply:
`StaticScopeGuard` is to track whether a statement is inside `ti.static`, and `NonStaticScopeGuard` is to track whether a statement is inside non-static `if`/`for`/`while`. Maybe we can rename `NonStaticScopeGuard` to `NonStaticControlFlowGuard` or something?

Reply:
Yep that will make things clearer.